### PR TITLE
change jade to pug, except extensions

### PR DESCRIPTION
**Pull request by andrewnot**
Hi!
As we have the problem of absence stable version of pug module (which is needed for apps in production), I've made changes on branche v1.x.x accordingly to the [comment](https://github.com/pugjs/pug/issues/2405#issuecomment-229907604).

So now, the module is - 'pug', service name is 'pug', but files will still have extension '.jade'.

Waiting for your feedback.
Thanks!

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -205,21 +205,21 @@ exports.compile = function(str, options){
...
       , '} catch (err) {'
-      , '  jade.rethrow(err, jade_debug[0].filename, jade_debug[0].lineno' + (options.compileDebug === true ? ',' + utils.stringify(str) : '') + ');'
+      , '  pug.rethrow(err, pug_debug[0].filename, pug_debug[0].lineno' + (options.compileDebug === true ? ',' + utils.stringify(str) : '') + ');'
       , '}'
     ].join('\n');
   } else {
     fn = parsed.body;
   }
-  fn = new Function('locals, jade', fn)
+  fn = new Function('locals, pug', fn)
```
Comment (TimothyGu) [inline lib/index.js]: I'm a bit worried that people are using internal symbols like `jade.escape` and `jade_mixins` themselves. Changing the parameter name might break these templates. @ForbesLindesay, what's your opinion on this?

Comment (ForbesLindesay) [inline lib/index.js]: They were completely un-documented, so I think it's OK with this still being a potential breakage when moving from `jade@1.x.x` to `pug@1.x.x`.  To be honest though, it's not something I have a strong opinion on.  Mostly I'm far more interested in getting people moved over to pug.

We only have 3 issues left before a stable release.  I'll try and look at some fo them over the weekend.

**Comment by sedja** [comment]
Hi, guys! 

Let me explain what is the main purpose of using this version of pug library.
There are a lot of old projects which are on maintance stage now. And we need to use the stable but not deprecated version of pug/jade module. 

For more convenience the extension left the same, so all we need is to change npm module name, instead of updating all views and code syntax. 

Please, tell your decision, for clarifying situation. 

Waiting for your feedback. Thanks

**Comment by sedja** [comment]
Sorry!
Closed it, by mistake

**Comment by TimothyGu** [comment]
Looks good to me.

**Comment by vdh** [comment]
When someone with npm access publishes this, remember to use something like
```sh
npm publish --tag stable
```
so that the 1.11.0 release is tagged for `pug@stable` and doesn't change the `pug@latest` tag away from 2.0

**Comment by sedja** [comment]
Hello!
Can you tell, when this release will be published?

**Comment by vdh** [comment]
@ForbesLindesay @jeromew Please, could someone get this merged and published? We really need a `pug@stable` / `pug@1.11.0` release.

**Comment by jeromew** [comment]
LGTM. I won't merge it myself because I don't have npm publishing rights + it would close this PR.

**Comment by nevech** [comment]
@TimothyGu @ForbesLindesay When you will have time to publish this release?

**Comment by zjrosen1** [comment]
Bump LGTM

**Pull request closed without merging**